### PR TITLE
Add `map::Value` imps for `bool` and `Option<T>`

### PR DESCRIPTION
**Pull request by korken89**
Added some more `Value` impls to reduce the needed boilerplate in users' crates.

### CODE REVIEW

File: src/map.rs [hunk 1]
```diff
@@ -803,6 +803,42 @@ impl<'a, const N: usize> Value<'a> for [u8; N] {
...
+        Self: Sized,
+    {
+        Ok(<u8 as Value>::deserialize_from(buffer)? != 0)
+    }
+}
+
+impl<'a, T: Value<'a>> Value<'a> for Option<T> {
+    fn serialize_into(&self, buffer: &mut [u8]) -> Result<usize, SerializationError> {
+        if let Some(val) = self {
+            <u8 as Value>::serialize_into(&1, buffer)?;
```
Comment (diondokter) [inline src/map.rs:822]: Maybe make this the fancy newfangled bool?

**Review by diondokter — CHANGES_REQUESTED** [review]
Cool, thanks! One thing and it's ready for merge

File: src/map.rs (hunk 1 above)
Comment (korken89) [reply src/map.rs]: Updated!

**Comment by korken89** [comment]
I also added an implementation for arrays.

**Comment by diondokter** [comment]
Alright, looks nice!

I almost thought impl'in the trait for bool would be a breaking change. But luckily it's not! Since bool is never a user-defined type and they couldn't have implemented it themselves.

**Comment by diondokter** [comment]
Oh! We now get that error in CI!

Weird! Maybe a different CI machine with an updated linker or something?
No idea. I'll merge anyways an submit an issue to figure it out.

**Comment by diondokter** [comment]
Ah! Actually these errors do need looking at.

See, your changes to the Value impls changes the way the type inference is done which messes up the tests.
And if it messes up the tests, then probably also user code, which would make this a breaking change.
I think a breaking change is fine for this, because it's better. But the tests do need to be fixed.

**Comment by korken89** [comment]
I'll fix the tests first thing in the morning! 🚀

**Comment by korken89** [comment]
I have update the tests and fixed the impl! 
It's a bit unfortunate that I cannot get the type inference to work the same which makes it into a breaking change. :disappointed:

**Comment by diondokter** [comment]
Thanks!

**Pull request merged**